=== tests/cases/conformance/internalModules/DeclarationMerging/class.ts ===
module X.Y {
>X : 类为 X
>Y : 类为 Y

    export class Point {
>Point : Point

        constructor(x: number, y: number) {
>x : 数字
>类型别名 :=> 数字
>y : 数字
>类型别名 :=> 数字

            this.x = x;
>this.x = x : 数字
>类型别名 :=> 数字
>this.x : 数字
>类型别名 :=> 数字
>this : 本体
>x : 数字
>类型别名 :=> 数字
>x : 数字
>类型别名 :=> 数字

            this.y = y;
>this.y = y : 数字
>类型别名 :=> 数字
>this.y : 数字
>类型别名 :=> 数字
>this : 本体
>y : 数字
>类型别名 :=> 数字
>y : 数字
>类型别名 :=> 数字
        }
        x: number;
>x : 数字
>类型别名 :=> 数字

        y: number;
>y : 数字
>类型别名 :=> 数字
    }
}

=== tests/cases/conformance/internalModules/DeclarationMerging/module.ts ===
module X.Y {
>X : 类为 X
>Y : 类为 Y

    export module Point {
>Point : 类为 Point

        export var Origin = new Point(0, 0);
>Origin : Point
>new Point(0, 0) : Point
>Point : 类为 Point
>0 : 0
>0 : 0
    }
}

=== tests/cases/conformance/internalModules/DeclarationMerging/test.ts ===
//var cl: { x: number; y: number; }
var cl = new X.Y.Point(1,1);
>cl : X.Y.Point
>new X.Y.Point(1,1) : X.Y.Point
>X.Y.Point : 类为 X.Y.Point
>X.Y : 类为 X.Y
>X : 类为 X
>Y : 类为 X.Y
>Point : 类为 X.Y.Point
>1 : 1
>1 : 1

var cl = X.Y.Point.Origin; // error not expected here same as bug 83996 ?
>cl : X.Y.Point
>X.Y.Point.Origin : X.Y.Point
>X.Y.Point : 类为 X.Y.Point
>X.Y : 类为 X.Y
>X : 类为 X
>Y : 类为 X.Y
>Point : 类为 X.Y.Point
>Origin : X.Y.Point


=== tests/cases/conformance/internalModules/DeclarationMerging/simple.ts ===
class A {
>A : A

    id: string;
>id : 文字
>类型别名 :=> 文字
}

module A {
>A : 类为 A

    export var Instance = new A();
>Instance : A
>new A() : A
>A : 类为 A
}

// ensure merging works as expected
var a = A.Instance;
>a : A
>A.Instance : A
>A : 类为 A
>Instance : A

var a = new A();
>a : A
>new A() : A
>A : 类为 A

var a: { id: string };
>a : A
>id : 文字
>类型别名 :=> 文字

